app/controllers/agents_controller.rb

```diff
@@ -75,6 +75,20 @@ def event_descriptions
     render :json => { :description_html => html }
   end
 
+  def reemit_events
+    @agent = current_user.agents.find(params[:id])
+
+    # `find_each` orders by PK, so events get re-created in the same order
+    @agent.events.find_each do |event|
+      event.reemit!
```

Comment:
Is this fast enough when an Agent has a lot of Events? I am wondering how many events can reemitted until the web request times out. It might be worth moving the action into a background job.

Reply:
I had the same thought, and knew someone might bring it up :) It's correct that in order to be scalable here, this should be moved into a background job.

One other thought I had was that if we bring it out to a job, we should probably pass along a timestamp to the job. My rationale is that you want the action to only affect events created _before_ you pushed that button (in case the job is stuck in a queue). Thoughts?

Honestly these are my first PRs here, and I wasn't sure if there would be even looked at or considered, so great to see this.

I will move this to a background job when I get a chance.

Reply:
> One other thought I had was that if we bring it out to a job, we should probably pass along a timestamp to the job. My rationale is that you want the action to only affect events created _before_ you pushed that button (in case the job is stuck in a queue). Thoughts?

That is a good point, especially for busy instances. Would passing in the latest Event id work as well? I'd imagine the query being faster than filtering by the `created_at` timestamp.

Reply:
You're right, that would be better.